optimum/habana/transformers/trainer.py

```diff
@@ -1872,6 +1799,26 @@ def evaluation_loop(
         # Will be useful when we have an iterable dataset so don't know its length.
         observed_num_examples = 0
 
+        # attn_softmax_bf16 and use_flash_attention are enabled only for llama, qwen2, starcoder2 and gemma
+        _should_update_inputs: bool = (getattr(self.model, "generation_config", None) is not None) and (
+            self.model.config.model_type in ["llama", "qwen2", "starcoder2", "gemma"]
+        )
+
+        if _should_update_inputs:
```

Comment:
Some of this code can be refactored, see other comment above about turning it into a function.